usecases/MinimumWorkingExample/dodo.py

```diff
@@ -374,3 +385,84 @@ def task_upload_to_openbis():
+        with open(exp_metadata_path) as file:
+            data = yaml.safe_load(file)
+        diameter = float(data['diameter'])
+        length = float(data['length'])
+
+        def calibrate_it(path_exp_data=processed_data_emodulus_directory,path_calibrated_data=calibrated_data_directory,
+                         exp_name=f):
+            output = read_exp_data_E_mod(path=path_exp_data, exp_name=exp_name + '.csv',
+                                         length=length, diameter=diameter)
+            E_samples = estimate_youngs_modulus(experimental_data=output,
+                                                calibration_metadata={"E_loc": E_loc, "E_scale": E_scale},
+                                                calibrated_data_path=path_calibrated_data, mode=config['mode'])
+            # store samples as csv
+            np.savetxt(os.path.join(calibrated_data_directory, f + '_calibrated_samples.csv'), E_samples, delimiter=',')
+        # current outputs
+        ## -- output from probeye. dont really need it but for the time being
+        owl_file = Path(calibrated_data_directory, 'calibrationWorkflow' + f + '.csv.owl')
+        displ_list = Path(calibrated_data_directory, 'displacement_list_' + f + '.csv.dat')
+        force_list = Path(calibrated_data_directory, 'force_list_' + f + '.csv.dat')
+        #another_file = Path(calibrated_data_directory, 'joint_samples_compression_test_calibration.dat')
+        ## -- created by the calibration script
+        calibrated_data_file = Path(calibrated_data_directory,f + '_calibrated_samples.csv')
+
+        yield {
+            'name': f'calibrate {f}',
+            'actions': [(calibrate_it,[processed_data_emodulus_directory,calibrated_data_directory,f])],
+            'file_dep': [exp_metadata_path,Path(processed_data_emodulus_directory,f+'.csv')], # the file dependancies, the metadata and exp files.
+            'targets': [owl_file,displ_list,force_list,calibrated_data_file], # the files which are output for each calibration
+            'clean': [clean_targets]
+        }
+@create_after(executed='perform_calibration')
+def task_perform_prediction():
+    # create folder, if it is not there
+    Path(predicted_data_directory).mkdir(parents=True, exist_ok=True)
+
+    # choose calibrated data to be used for prediction
+    calibrated_data = single_example_name
+    calibrated_data_path = os.path.join(calibrated_data_directory,calibrated_data+'_calibrated_samples.csv')
+    df = pd.read_csv(calibrated_data_path, header=None)
+    samples = df[0].tolist()
+
+    # output of this step
+    predicted_data_path = os.path.join(predicted_data_directory, 'prediction_with_' + calibrated_data + '.csv')
+
+    def do_prediction(samples):
+        # perform prediction
+        pos_pred = perform_prediction(forward_solver=wrapper_three_point_bending,parameter=samples,mode=config['mode'])
+
+        # store prediction in csv
+
+        np.savetxt(predicted_data_path,pos_pred,delimiter=',')
```

Comment:
I do not understand the output of this.
Currently the output for the one implemented example is nine time the same number.
I would have expected the samples to have the same number as the sample output of the calibration.
In addition, I do not understand why the output is always the same, should it not slightly change?

Reply:
thanks for pointing out. will have a look. May have some silly mistakes.

Reply:
fixed

Reply:
btw for cheap I defined "using 5 samples from the pool of samples. so you get 5 outputs of the posterior.